include/swift/AST/Decl.h

```diff
@@ -3787,22 +3783,62 @@ class ClassDecl final : public NominalTypeDecl {
   }
 
   Optional<bool> getCachedInheritsSuperclassInitializers() const {
-    if (Bits.ClassDecl.ComputedInheritsSuperclassInits)
-      return Bits.ClassDecl.InheritsSuperclassInits;
+    bool hasAttr =
+        getAttrs().hasAttribute<InheritsConvenienceInitializersAttr>();
+    if (hasAttr || Bits.ClassDecl.ComputedInheritsSuperclassInits)
+      return hasAttr;
+
+    return None;
+  }
+
+  Optional<bool> getCachedHasMissingDesignatedInitializers() const {
+    if (!Bits.ClassDecl.ComputedHasMissingDesignatedInitializers) {
+      // Force loading all the members, which will add this attribute if any of
+      // members are determined to be missing while loading.
+      auto mutableThis = const_cast<ClassDecl *>(this);
+      (void)mutableThis->lookupDirect(DeclBaseName::createConstructor());
+    }
+
+    bool hasAttr = getAttrs().hasAttribute<HasMissingDesignatedInitsAttr>();
+    if (hasAttr || Bits.ClassDecl.ComputedHasMissingDesignatedInitializers)
+      return hasAttr;
 
     return None;
   }
 
+
+  void setHasMissingDesignatedInitializers(bool value) {
+    auto existing = getAttrs()
+        .getAttribute<HasMissingDesignatedInitsAttr>();
+    if (value && !existing) {
+      auto attr = new (getASTContext())
```

Comment:
Lets not synthesize new attributes here. Instead, serialize the bit directly.

Reply:
We need to print the attribute in the module interface

Reply:
Can you handle that entirely in the ASTPrinter or would it be too messy? We have similar issues with @objc IIRC.

Reply:
That's a good point, I can move back to this being a bit and then the attribute can just override the bit.

Reply:
Ah, no, we want to diagnose whether this attribute was added or removed, because it changes ABI. I'd rather keep this as an attribute.

Reply:
Though, I suppose those can just become semantic checks in the api digester